geopandas/tests/test_dissolve.py

```diff
@@ -47,18 +47,30 @@ def test_geom_dissolve(self):
 
     def test_first_dissolve(self):
         test = self.polydf.dissolve('manhattan_bronx')
-        test = test.drop('myshapes', axis=1)
-        first = self.first.drop('myshapes', axis=1)
-        assert_frame_equal(first, test)
+        test.columns = test.columns.astype('str')
+        assert_frame_equal(self.first, test)
 
     def test_mean_dissolve(self):
         test = self.polydf.dissolve('manhattan_bronx', aggfunc='mean')
-        test = test.drop('myshapes', axis=1)
-        mean = self.mean.drop('myshapes', axis=1)
-        assert_frame_equal(mean, test)
+        test.columns = test.columns.astype('str')
+        assert_frame_equal(self.mean, test)
 
         test = self.polydf.dissolve('manhattan_bronx', aggfunc=np.mean)
-        test = test.drop('myshapes', axis=1)
-        assert_frame_equal(mean, test)
-
-
+        test.columns = test.columns.astype('str')
```

Comment:
@jorisvandenbossche note this is where I moved the column type-casts. Don't know why, but in python 2.7, the columns keep coming out of `dissolve` as type `mixed`, while the hand-crafted comparision has type `string`, which throws off `assert_frame_equal`.